python/pyarrow/compute.py

```diff
@@ -597,16 +597,29 @@ def field(name_or_index):
     Stores only the field's name. Type and other information is known only when
     the expression is bound to a dataset having an explicit scheme.
 
+    Nested references are allowed by passing a tuple of names.
+    For example ``('foo', 'bar')`` references the field named "bar" inside
+    the field named "foo".
+
     Parameters
     ----------
-    name_or_index : string or int
-        The name or index of the field the expression references to.
+    name_or_index : string, tuple or int
```

Comment:
I wonder if we should also allow dotted names (though that could get ambiguous), or else, allow `field("a", "b")` instead of requiring `field(("a", "b"))`. That said neither are essential here.

Reply:
I think that allowing `field("a", "b")` would make a lot of sense. Will try to implement it here and if I run into difficulties I will add a follow-up JIRA.